internal/app/agent/register/application.go

```diff
@@ -0,0 +1,62 @@
+package register
+
+import (
+	"sync"
+	"time"
+)
+
+type Status string
+
+const (
+	timeout  = 3
+	interval = 10000
+
+	Online  Status = "online"
+	Offline Status = "offline"
+)
+
+type Application struct {
```

Comment:
I am not sure that this should be an `Application` and not a `Service`. We need to brainstorm and find a proper name, because this relates to all that can be deployed on the gateway - Mainflux services, 3rd party infrastructural services, or 3rd party applications (like some analytics for example)...

Reply:
for now it is applications no more services , when decided lets search and replace

Reply:
I am not convinced.